Jenkinsfile

```
@@ -75,114 +110,136 @@ stage('Build') {
                     }
                 }
             }
-        }
-    }
 
-    if (!infra.isTrusted()) {
-        def images = [
-                'almalinux_jdk11',
-                'alpine_jdk11',
-                'alpine_jdk17',
-                'alpine_jdk21',
-                'debian_jdk11',
-                'debian_jdk17',
-                'debian_jdk21',
-                'debian_slim_jdk11',
-                'debian_slim_jdk17',
-                'debian_slim_jdk21',
-                'rhel_ubi8_jdk11',
-                'rhel_ubi9_jdk17',
-                'rhel_ubi9_jdk21',
-        ]
-        for (i in images) {
-            def imageToBuild = i
-
-            builds[imageToBuild] = {
-                nodeWithTimeout('docker') {
-                    deleteDir()
-
-                    stage('Checkout') {
-                        checkout scm
-                    }
+            if (!infra.isTrusted()) {
+                def images
+
+                def imagesWithoutJava11 = [
+                    'alpine_jdk17',
+                    'alpine_jdk21',
+                    'debian_jdk17',
+                    'debian_jdk21',
+                    'debian_slim_jdk17',
+                    'debian_slim_jdk21',
+                    'rhel_ubi9_jdk17',
+                    'rhel_ubi9_jdk21',
+                ]
+                def imagesWithJava11 = [
+                    'almalinux_jdk11',
+                    'alpine_jdk11',
+                    'alpine_jdk17',
+                    'alpine_jdk21',
+                    'debian_jdk11',
+                    'debian_jdk17',
+                    'debian_jdk21',
+                    'debian_slim_jdk11',
+                    'debian_slim_jdk17',
+                    'debian_slim_jdk21',
+                    'rhel_ubi8_jdk11',
+                    'rhel_ubi9_jdk17',
+                    'rhel_ubi9_jdk21',
+                ]
+                if (isLTS) {
+                    images = imagesWithJava11
+                } else {
+                    images = imagesWithoutJava11
+                }
+                // Build all images including Java 11 if the version match a LTS versioning pattern
+                // TODO: remove when Java 11 is removed from LTS line
+                // See https://github.com/jenkinsci/docker/issues/1890
```

```suggestion
                // Build all images including Java 11 if the version match a LTS versioning pattern
                // TODO: remove when Java 11 is removed from LTS line
                // See https://github.com/jenkinsci/docker/issues/1890
                if (isLTS) {
                    images = imagesWithJava11
                } else {
                    images = imagesWithoutJava11
                }
```
(Comments were meant to be above sorry, will commit this change after the first review feedbacks)